overlord/ifacestate/handlers.go

```diff
@@ -653,6 +653,17 @@ func checkAutoconnectConflicts(st *state.State, plugSnap, slotSnap string) error
 
 		// other snap that affects us because of plug or slot
 		if k == "unlink-snap" || k == "link-snap" || k == "setup-profiles" || k == "discard-snap" {
+			if k == "setup-profiles" {
```

Comment:
(nitpick^2) I think I would extract this into a helper to make the flow of the function easier to read. Maybe even right after the `// different snaps - no conflict` if. Something like:
```
               // different snaps - no conflict
 		if otherSnapName != plugSnap && otherSnapName != slotSnap {
 			continue
                }
                if obsoleteCorePhase2SetupProfiles(k, task) {
                           continue
                }

                // other snap that affects us because of plug or slot
		if k == "unlink-snap" || k == "link-snap" || k == "setup-profiles" || k == "discard-snap" {
...


```